src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -64,10 +64,51 @@ void parallel_reduce (const TeamMember& team,
 
    result = local_tmp;
   } else {
-    Kokkos::parallel_reduce(Kokkos::TeamThreadRange(team, begin, end), lambda, result);
+    Kokkos::parallel_reduce(loop_boundaries, lambda, result);
   }
 }
 
+template <bool Serialize, int TotalSize, int VectorSize, typename TeamMember, typename InputProvider, typename ValueType>
+static KOKKOS_INLINE_FUNCTION
+void view_reduction (const TeamMember& team,
+                     const int& begin,
+                     const int& end,
+                     const InputProvider& input,
+                     ValueType& result)
+{
+  if (Serialize) {
+    const int  n_packs        = end - begin;
+    const bool has_garbage    = (TotalSize % VectorSize == 0 ? false : true);
+    const int  last_pack_size = (has_garbage ? TotalSize % VectorSize : VectorSize);
+    const int  loop_length    = (has_garbage ? n_packs-1 : n_packs);
+
+    impl::parallel_reduce<Serialize>(team,Kokkos::ThreadVectorRange(team, loop_length),
```

Comment:
If begin!=0, then this would be buggy, since the || reduce would be in `[0,loop_length)`. I think you need to explicilty construct the TVR passing `[begin, begin+loop_length)`.

Reply:
Also, why is this on a TVR and not a TTR?

Reply:
This could be a misunderstanding of Kokkos, but this was the last level of parallelism, so I chose TVR.

Reply:
Perhaps it will help if I point out that this is operating at exactly the ||ism level that the reductions (and all the k loops) need to be done in SHOC.

Reply:
There is now new begin/end logic and I create a test case for each scenario.